src/Aspire.Cli/Aspire.Cli.csproj

```diff
@@ -31,6 +31,10 @@
     <PublishAot>false</PublishAot>
   </PropertyGroup>
 
+  <ItemGroup Condition="'$(PublishAot)' == 'true'">
+    <LinkerArg Condition="'$(RuntimeIdentifier)' == 'linux-musl-x64'" Include="--target=x86_64-alpine-linux-musl" />
```

Comment:
Why is this necessary? This doesn't seem right that we have to do this.

cc @agocke @MichalStrehovsky

Reply:
IIUC, this is fixed in net9.0 but Aspire.Cli targets net8.0 . cc @akoeplinger

Reply:
We should find a time to update to `net10.0`. I started in https://github.com/dotnet/aspire/pull/10167. Maybe when .NET 10 RC (or maybe preview7?) comes out, it would be a good time to switch to the latest. Advantages:

1. I noticed a decent disk size drop when I was trying it. 
2. It would allow us to do "warnings as errors", which was also fixed in net9.0.

Reply:
> Why is this necessary? This doesn't seem right that we have to do this.

FWIW this was fixed in https://github.com/dotnet/runtime/pull/101213, looks like there's an LLVM bug asking why this is needed.

Reply:
> Would that mean that we could no longer publish a dotnet tool that is installable in 8 or 9 SDKs though?

Yes, or at minimum you could publish a 8/9 to but it would need to be a different package id.